src/main/java/br/com/sbs/cubatech/course/CourseRepository.java

```diff
@@ -21,4 +21,5 @@
 
     Page<Course> findAllBySubCategory(SubCategory subCategory, Pageable pageable);
 
+    Course findByUrlCode(String courseCode);
```

Comment:
Daria pra colocar um `Optional` que o Spring Data JPA resolve.

```suggestion
    Optional<Course> findByUrlCode(String courseCode);
```